ml_server/main.py

```diff
@@ -0,0 +1,79 @@
+import os
+from pathlib import Path
+from fastapi import FastAPI, HTTPException
+from pydantic import BaseModel
+from typing import List
+from functools import lru_cache
+
+from fastapi import Depends
+from gliner import GLiNER
+
+from tenant_dependency import get_tenant_info
+
+KEYCLOAK_HOST = os.getenv("KEYCLOAK_HOST", "")
+TENANT = get_tenant_info(url=KEYCLOAK_HOST, algorithm="RS256")
+ROOT_PATH = os.environ.get("ROOT_PATH", default="")
+
+
+# Cache will save up to this number of most recently used models. Larger amount will consume more memory,
+# but allows more models to be initialized at the same time.
+NUM_MODELS_TO_CACHE = 3
+
+def get_version() -> str:
+    default = "0.1.0"
+    ver = Path(__file__).parent.parent / "version.txt"
+    if ver.exists() and ver.is_file():
+        with open(ver, "r", encoding="utf-8") as file:
+            line = file.readline().strip()
+            return line or default
+
+    return default
+
+app = FastAPI(
+    title="ML models hosting server",
+    version=get_version(),
+    root_path=ROOT_PATH,
+    dependencies=[Depends(TENANT)])
+
+# Request body model
+class PredictionRequest(BaseModel):
+    model_id: str
+    threshold: float
+    labels: List[str]
+    text: str
+
+
+@lru_cache(maxsize=NUM_MODELS_TO_CACHE)
+def get_model(model_id: str) -> GLiNER:
+    """Download and initialise specified ML model
+    
+    Args:
+        model_id (str): ID of the model to download
+
+    Returns:
+        GLiNER: initialized Gliner model
+    """
+    try:
+        # Initialize and cache the model
+        model = GLiNER.from_pretrained(model_id)
+    except Exception as e:
```

Comment:
Can we except more specific exception? This one is too general. Additionally would be nice to have more explanatory http exception. Something like "Could not load model model_id=...".
The one that comes to mind when you haven't found the model. And it should be 404 in this case.
Don't expose exception to outside. Rather raise http exception from caught one using dedicated syntax

Reply:
I didn't want to spend time going and searching the source code of the library for all possible exceptions that Gliner can raise. That's why I used general exception, basically you still will get info about original exception in the output, and for us as a user of that library the most important part is that it's failed and we can't proceed.